### PR TITLE
components: Remove `@emotion/css` from Spacer

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Part of https://github.com/WordPress/gutenberg/issues/30503#deprecating-emotion-css

I also added a story to make it easier to test.

## How has this been tested?
Test it out in Storybook. All the props should work as expected.

## Types of changes
Non breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/spacer/styles.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { InnerProps } from './types';
+import { space } from '../ui/utils/space';
+
+const isDefined = < T >( o: T ): o is Exclude< T, null | undefined > =>
+	typeof o !== 'undefined' && o !== null;
```
Comment (ciampo) [inline packages/components/src/spacer/styles.ts:16]: Neat!

File: packages/components/src/spacer/styles.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	${ renderProperty( 'margin' ) }
+
+	${ renderProperty( 'paddingTop' ) }
+	${ renderProperty( 'paddingBottom' ) }
+	${ renderProperty( 'paddingLeft' ) }
+	${ renderProperty( 'paddingRight' ) }
+	${ renderProperty( 'paddingX', [ 'paddingLeft', 'paddingRight' ] ) }
+	${ renderProperty( 'paddingY', [ 'paddingTop', 'paddingBottom' ] ) }
+	${ renderProperty( 'padding' ) }
+`;
```
Comment (ciampo) [inline packages/components/src/spacer/styles.ts]: Currently the more "generic" properties, when set, override the more "specific" properties.

If we swapped the order, we would enable scenarios where the more generic and the more specific properties could be defined at the same time (e.g. `padding` and `padding-left`

```suggestion
export const SpacerView = styled.div< InnerProps >`
	${ renderProperty( 'margin' ) }
	${ renderProperty( 'marginY', [ 'marginTop', 'marginBottom' ] ) }
	${ renderProperty( 'marginX', [ 'marginLeft', 'marginRight' ] ) }
	${ renderProperty( 'marginTop' ) }
	${ renderProperty( 'marginBottom' ) }
	${ renderProperty( 'marginLeft' ) }
	${ renderProperty( 'marginRight' ) }

	${ renderProperty( 'padding' ) }
	${ renderProperty( 'paddingY', [ 'paddingTop', 'paddingBottom' ] ) }
	${ renderProperty( 'paddingX', [ 'paddingLeft', 'paddingRight' ] ) }
	${ renderProperty( 'paddingTop' ) }
	${ renderProperty( 'paddingBottom' ) }
	${ renderProperty( 'paddingLeft' ) }
	${ renderProperty( 'paddingRight' ) }
`;
```

File: packages/components/src/spacer/stories/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export const _default = () => {
+	const props = PROPS.reduce(
+		( acc, prop ) => ( { ...acc, [ prop ]: number( prop, undefined ) } ),
+		{}
+	);
+
+	return (
+		<>
+			<BlackBox />
+			<Spacer { ...props }>This is the spacer</Spacer>
```
Comment (ciampo) [inline packages/components/src/spacer/stories/index.js:49]: Thank you for adding a story!

**Review by ciampo — APPROVED** [review]
I just left a comment about the order in which CSS props are applied, but otherwise LGTM!

**Review by ciampo — COMMENTED** [review]
Changing the order in which the `margin` and `padding` props are applied caused a snapshot test "error", which highlighted a side effect of this change.

Apart from updating the snapshot, we should also look for every usage of `<Spacer />` and  see if the `margin`/`marginX`/`marginY`/`padding`/`paddingX`/`paddingY` props are specified and are expected to override any "single" top/right/bottom/left value — as that wouldn't be the case anymore with the changes from this PR.

**Comment by sarayourfriend** [comment]
> we should also look for every usage of <Spacer />

Lucky there are no usages of the component 🙂

**Pull request closed without merging**